File: solutions/899/main.cpp
#include <string> 

class Solution {
public: 
    string orderlyQueue(string s, int k) {
        int n = s.size(); 
        if (k == 1) {
            string ans = s; 
            for (int i = 1; i < n; i++) {
                string t = s.substr(i) + s.substr(0, i); 
                if (ans > t) ans = t; 
            }
            return ans; 
        }
        sort(s.begin(), s.end());
        return s;
    }
};

File: solutions/899/main.rs
impl Solution {
    pub fn orderly_queue(s: String, k: i32) -> String {
        let mut v = s.chars().collect::<Vec<char>>(); 
        let n = v.len(); 

        if k == 1 {
            // ref https://doc.rust-lang.org/std/vec/struct.Vec.html#method.extend_from_within
            v.extend_from_within(0..); 
            let mut ans = &v[0..n]; 

            for i in 1..n {
                if ans > &v[i..n+i] {
                    ans = &v[i..n+i]; 
                }
            }

            ans.into_iter().collect()
        } else {
            v.sort(); 
            v.into_iter().collect()
        }
    }
}